python/cudf/cudf/core/column/column.py

```diff
@@ -1323,7 +1323,9 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
     elif np.isscalar(arbitrary) and not isinstance(arbitrary, memoryview):
         length = length or 1
         data = as_column(
-            utils.scalar_broadcast_to(arbitrary, length, dtype=dtype)
+            utils.scalar_broadcast_to(
+                arbitrary, length, dtype=dtype, nan_as_null=nan_as_null
+            )
```

Comment:
Instead of pushing the conversion down into the `scalar_broadcast_to` function, we should convert the scalar passed into the function based on the `nan_as_null` parameter.